src/ConnectedMode/UI/Credentials/CredentialsViewModel.cs

```diff
@@ -0,0 +1,108 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.ObjectModel;
+using System.IO;
+using SonarLint.VisualStudio.ConnectedMode.UI.Resources;
+using SonarLint.VisualStudio.Core.WPF;
+using static SonarLint.VisualStudio.ConnectedMode.ConnectionInfo;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UI.Credentials;
+
+public class CredentialsViewModel(Connection connection) : ViewModelBase
+{
+    private string token;
+    private string selectedAuthenticationType = UiResources.Token;
+    private string username;
+    private string password;
+
+    public Connection Connection { get; } = connection;
+    public ObservableCollection<string> AuthenticationType { get; } = [UiResources.Token, UiResources.Credentials];
+
+    public string SelectedAuthenticationType
+    {
+        get => selectedAuthenticationType;
+        set
+        {
+            selectedAuthenticationType = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsTokenAuthentication));
+            RaisePropertyChanged(nameof(IsCredentialsAuthentication));
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldTokenBeFilled));
+            RaisePropertyChanged(nameof(ShouldUsernameBeFilled));
+            RaisePropertyChanged(nameof(ShouldPasswordBeFilled));
+        }
+    }
+
+    public string Token
+    {
+        get => token;
+        set
+        {
+            token = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldTokenBeFilled));
+        }
+    }
+
+    public string Username
+    {
+        get => username;
+        set
+        {
+            username = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldUsernameBeFilled));
+        }
+    }
+
+    public string Password
+    {
+        get => password;
+        set
+        {
+            password = value;
+            RaisePropertyChanged();
+            RaisePropertyChanged(nameof(IsConfirmationBtnEnabled));
+            RaisePropertyChanged(nameof(ShouldPasswordBeFilled));
+        }
+    }
+
+    public bool IsTokenAuthentication => SelectedAuthenticationType == UiResources.Token;
+    public bool IsCredentialsAuthentication => SelectedAuthenticationType == UiResources.Credentials;
+    public bool ShouldTokenBeFilled => IsTokenAuthentication && !IsTokenProvided;
+    public bool ShouldUsernameBeFilled => IsCredentialsAuthentication && !IsUsernameProvided;
+    public bool ShouldPasswordBeFilled => IsCredentialsAuthentication && !IsPasswordProvided;
+    public bool IsConfirmationBtnEnabled => (IsTokenAuthentication && IsTokenProvided) || (IsCredentialsAuthentication && AreCredentialsProvided);
```

Comment:
I would not use contractions like Btn. I think in this case you may be able to just call it `IsConfirmationEnabled` since it's obvious from the context that there's only one type of confirmation